src/ring_theory/int/basic.lean

```diff
@@ -341,3 +341,55 @@ instance decidable_int : decidable_rel (λ a b : ℤ, (multiplicity a b).dom) :=
 λ a b, decidable_of_iff _ finite_int_iff.symm
 
 end multiplicity
+
+lemma induction_on_primes {P: nat → Prop} (h₁: P 0) (h₂: P 1)
+  (h: ∀ p a: ℕ, nat.prime p → P a → P (p * a)): ∀ n: ℕ, P n :=
+begin
+  intro n,
+  apply unique_factorization_monoid.induction_on_prime,
+  exact h₁,
+  { intros n h,
+    rw nat.is_unit_iff.1 h,
+    exact h₂, },
+  { intros a p _ hp ha,
+    exact h p a (nat.prime_iff_prime.2 hp) ha, },
+end
+
+lemma int.associated_nat_abs (k: ℤ): associated k k.nat_abs :=
```

Comment:
A general style comment: mathlib always puts spaces on both sides of `:`
```suggestion
lemma int.associated_nat_abs (k : ℤ) : associated k k.nat_abs :=
```
This applies throughout this file, and to your other PRs too.